presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -592,6 +603,21 @@ public void enforceTimeLimits()
         }
     }
 
+    /**
+     * Enforce query CPU time limits
+     */
+    public void enforceCpuLimits()
+    {
+        for (QueryExecution query : queries.values()) {
+            Duration cpuTime = query.getTotalCpuTime();
+            Duration sessionLimit = getQueryMaxCpuTime(query.getSession());
+            Duration limit = maxQueryCpuTime.compareTo(sessionLimit) < 0 ? maxQueryCpuTime : sessionLimit;
```

Comment:
`Duration limit = Ordering.natural().min(maxQueryCpuTime, sessionLimit)` would highlight that you take lower of two.